assets/js/login.js

```diff
@@ -0,0 +1,35 @@
+import { toast } from "./toast.js";
+import ApiHandler from "./ApiHandler.js";
+
+const api = new ApiHandler("https://py-library-api-v2.server.steffen.codes");
+
+const loginForm = document.getElementById("loginForm");
+
+loginForm.addEventListener("submit", async (e) => {
+  e.preventDefault();
+
+  const submitButton = loginForm.querySelector("button[type=submit]");
+  const formData = new FormData(loginForm);
+  const data = Object.fromEntries(formData.entries());
+
+  submitButton.disabled = true;
+  const loginResult = await api.loginUser(data.email.trim(), data.password);
+  if (loginResult.success) {
+    localStorage.setItem("userId", data.email.trim());
+    localStorage.setItem("email", loginResult.data.user_id);
+    toast("Login successful, navigating you to the frontpage ..", "success");
+    setTimeout(() => (window.location.href = "/index.html"), 2000);
+  } else {
+    submitButton.disabled = false;
+    switch (loginResult.statusCode) {
+      case 400:
+        toast("Missing e-mail or password", "error");
```

Comment:
I dont think the toasts here are being raised due to the pattern attribute in the html form. 
So i think we should either handle all validation in javascript or delete the toast entirely and use the pattern attribute and browser-specific alerts in all cases (login, signup, all other forms). 

Not sure which i best but i have a feeling Arturo might like this approach better since the html structure is more correct haha

Reply:
The toast aren't (or shouldn't) be raised no, the validation is already done, so no parameters should be missing. But we should account for all status codes, that the endpoint might respond with.

I like this style of validation with patterns, it isn't something I've used before, so I'm quite unfamiliar with it, but it cuts down on the Javascript side of things a lot.